tests/module/memory_test.c

```diff
@@ -0,0 +1,191 @@
+    param.algo = algo;
+    switch (algo) {
+    case VecSimAlgo_BF:
+        param.bfParams.blockSize = 1;
+        param.bfParams.initialCapacity = 1;
+        break;
+
+    case VecSimAlgo_HNSWLIB:
+        param.hnswParams.M = 30;
+        param.hnswParams.initialCapacity = 1;
+        param.hnswParams.efConstruction = 0;
+        param.hnswParams.efRuntime = 0;
+        break;
+    }
+    param.type = VecSimType_INT64;
+    param.size = DIMENSION;
+    param.metric = VecSimMetric_L2;
+
+    return VecSimIndex_New(&param);
+}
+
+// creates an index, add n vectors, deletes m vectors and then compares redis-server memory usage
+// report to the index memory usage report uncomment the 3 lines close to the end to get the values
+// along with the comparison result.
+int VecSim_memory_create_index_add_n_delete_m_check(RedisModuleCtx *ctx, RedisModuleString **argv,
+                                                    int argc) {
+
+    long long n, m, i;
+    long long startMemory, endMemory;
+    VecSimAlgo algo = -1;
+    RedisModuleString *al;
+
+    if (argc != 4 || RedisModule_StringToLongLong(argv[2], &n) == REDISMODULE_ERR ||
+        RedisModule_StringToLongLong(argv[3], &m) == REDISMODULE_ERR) {
+        RedisModule_WrongArity(ctx);
+        return REDISMODULE_OK;
+    }
+    al = RedisModule_CreateString(ctx, "BF", 2);
+    if (!RedisModule_StringCompare(argv[1], al))
+        algo = VecSimAlgo_BF;
+    RedisModule_FreeString(ctx, al);
+    al = RedisModule_CreateString(ctx, "HNSW", 4);
+    if (!RedisModule_StringCompare(argv[1], al))
+        algo = VecSimAlgo_HNSWLIB;
+    RedisModule_FreeString(ctx, al);
+    if (algo == -1) {
+        RedisModule_WrongArity(ctx);
+        return REDISMODULE_OK;
+    }
+
+    startMemory = _get_memory_usage(ctx); // get memory usage before creating the index
+
+    VecSimIndex *index = _create_index(algo);
+    _add_vectors(index, n);
+    _delete_vectors(index, m);
+
+    endMemory = _get_memory_usage(ctx); // get memory usage after creating the index
+
+    VecSimIndexInfo indexInfo = VecSimIndex_Info(index);
+
+    // RedisModule_ReplyWithArray(ctx, 3);
+    // RedisModule_ReplyWithLongLong(ctx,endMemory-startMemory);
+    // RedisModule_ReplyWithLongLong(ctx,indexInfo.memory);
+
+    if (indexInfo.memory <= endMemory - startMemory)
+        RedisModule_ReplyWithSimpleString(ctx, "OK");
+    else
+        RedisModule_ReplyWithError(ctx, "ERROR");
+
+    VecSimIndex_Free(index);
+    return REDISMODULE_OK;
+}
+
+// calls to VecSim_memory_create_index_add_n_delete_m_check with m=0
+int VecSim_memory_create_index_add_n_check(RedisModuleCtx *ctx, RedisModuleString **argv,
+                                           int argc) {
+    int i, res;
+    RedisModuleString *newArgv[argc + 1];
+    for (i = 0; i < argc; i++)
+        newArgv[i] = argv[i];
+    newArgv[argc] = RedisModule_CreateString(ctx, "0", 1);
+    res = VecSim_memory_create_index_add_n_delete_m_check(ctx, newArgv, argc + 1);
```

Comment:
I see what you meant here, but try do it cleaner. Consider having 
`VecSim_memory_check_impl` or something similar, where you can call it from both `VecSim_memory_create_index_add_n_delete_m_check` and `VecSim_memory_create_index_add_n_check` after you parsed `argv` and without creating new `RedisModuleString`